src/protocol/coap.rs

```diff
@@ -0,0 +1,191 @@
+//! CoAP Binding Template
+
+use crate::extend::ExtendableThing;
+use serde::{Deserialize, Serialize};
+use serde_repr::{Deserialize_repr, Serialize_repr};
+use serde_with::{serde_as, skip_serializing_none};
+
+/// CoAP request method
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum Method {
+    Get,
+    Put,
+    Post,
+    Delete,
+    Patch,
+    Fetch,
+    #[serde(rename = "iPATCH")]
+    Ipatch,
+}
+
+/// CoAP Allowed block size
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Deserialize_repr, Serialize_repr)]
+#[repr(u16)]
+pub enum BlockSize {
+    Szx16 = 16,
+    Szx32 = 32,
+    Szx64 = 64,
+    Szx128 = 128,
+    Szx256 = 256,
+    Szx512 = 512,
+    Szx1024 = 1024,
+}
+
+/// CoAP BlockWise Transfer Parameters
+///
+/// They may apply to Block-Wise Transfers [RFC7959] or
+/// Block-Wise Transfer Options Supporting Robust Transmission [RFC9177].
+///
+/// [RFC7959]: https://www.rfc-editor.org/rfc/rfc7959.html
+/// [RFC9177]: https://www.rfc-editor.org/rfc/rfc9177.html
+#[serde_as]
+#[skip_serializing_none]
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash, Default)]
```

Comment:
```suggestion
#[derive(Debug, Clone, Copy, Deserialize, Serialize, PartialEq, Eq, Hash, Default)]
```

Reasoning: `BlockSize` is 16bits, becomes 32bits if the compiler is not smart enough to use a niche, the entire struct should then fill in 64 bits.